src/frontend/js/components/CourseProductItem/index.tsx

```diff
@@ -32,16 +33,17 @@ export interface Props {
 const CourseProductItem = ({ productId, courseCode }: Props) => {
   const productQuery = useProduct(productId, { course: courseCode });
   const product = productQuery.item;
-  const isOwned = useMemo(() => product?.orders?.length > 0, [productQuery.item]);
-  const { item: order, ...orderQuery } = useOrder(
-    product?.orders?.[0],
-    {
-      state: [OrderState.VALIDATED],
-    },
-    {
-      enabled: isOwned,
-    },
+  const ordersQuery = useOrders({
+    product: productId,
+    course: courseCode,
+    state: [OrderState.VALIDATED, OrderState.PENDING],
```

Comment:
Why are you including `OrderState.PENDING` is the filters while, below at the line `42` you filter again to find the first `OrderState.VALIDATED` order? Why not this?


```suggestion
  const ordersQuery = useOrders({
    product: productId,
    course: courseCode,
    state: [OrderState.VALIDATED],
  });
```

Reply:
Because `isOwned` flag must be true if the user has a validated or a pending order. In this way, the user is not allow to open sale tunnel if it has a pending order.

But user should be allowed to enroll to course run only if it has a validated order.

Reply:
Oh! The `isOwned` was wrong! I added the test to test the behavior I described above.